### PR TITLE
Ngrok

### PR DESCRIPTION
**Pull request by gajendraJangid421**

### CODE REVIEW

File: Event-Management-Application/event_management/pom.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-security</artifactId>
+		</dependency>
+
+		<!--		<dependency>-->
+<!--			<groupId>com.github.ulisesbocchio</groupId>-->
+<!--			<artifactId>jasypt-spring-boot-starter</artifactId>-->
+<!--			<version>3.0.5</version>-->
+<!--		</dependency>-->
```
Comment (sagarzond) [inline Event-Management-Application/event_management/pom.xml:89]: remove commented line

File: Event-Management-Application/event_management/pom.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		<plugins>
+			<plugin>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+			</plugin>
+<!--			<plugin>-->
+<!--				<groupId>com.github.ulisesbocchio</groupId>-->
+<!--				<artifactId>jasypt-maven-plugin</artifactId>-->
+<!--				<version>3.0.5</version>-->
+<!--			</plugin>-->
```
Comment (sagarzond) [inline Event-Management-Application/event_management/pom.xml:104]: remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/EventManagementApplication.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class EventManagementApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(EventManagementApplication.class, args);
+
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/EventManagementApplication.java:11]: extra line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import com.example.event_management.service.EventsService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "api/events")
+@CrossOrigin(origins = "https://localhost:3000")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:15]: This is not required as you have disabled CORS in another file.

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+public class EventsController {
+
+    @Autowired
+    private EventsService eventsService;
+
+    //for admin and user
+    @GetMapping(path = "")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:22]:
```suggestion
    @GetMapping
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    public List<Events> retrieveAllEventsList() {
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:23]:
```suggestion
    public List<Events> findAll() {

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        return eventsService.findAll();
+    }
+
+    //for admin
+    @GetMapping(path = "/{id}")
+    public Events retrieveEventById(@PathVariable String id){
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:29]:
```suggestion
    public Events findById(@PathVariable String id){

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        return eventsService.findById(id);
+    }
+
+//    //for admin and user
+//    @GetMapping(path = "/event-name/{eventName}")
+//    public Events retrieveEventByEventName(@PathVariable String eventName){
+//        return eventsService.findByName(eventName);
+//    }
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:37]: remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+    //for admin
+    @PostMapping(path = "/add-event")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:40]: according to REST API standards.
```suggestion
    @PostMapping
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    public Events save(@Valid @RequestBody Events events){
+        return eventsService.save(events);
+    }
+
+    //for admin
+    @PutMapping(path = "/{id}")
+    public Events update(@PathVariable String id, @Valid @RequestBody Events events){
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:47]:
```suggestion
    public Events updateById(@PathVariable String id, @Valid @RequestBody Events events){

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+        return eventsService.update(id, events);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:48]:
```suggestion
        events.setId(id);
        return eventsService.update(events);

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    //for admin
+    @DeleteMapping(path = "/{id}")
+    public void delete(@PathVariable String id){
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:53]:
```suggestion
    public void deleteById(@PathVariable String id){

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        eventsService.delete(id);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:54]:
```suggestion
        eventsService.deleteById(id);

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/LoginController.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+import com.example.event_management.model.Session;
+import com.example.event_management.service.LoginService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(path = "/api/users")
+@CrossOrigin(origins = "https://localhost:3000")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/LoginController.java:14]: not required

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/LoginController.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+public class LoginController {
+
+    @Autowired
+    LoginService loginService;
+
+    //for user
+    @PostMapping(path = "/login")
+    public LoginResponse userLogin(@Valid @RequestBody LoginRequest loginRequest) {
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/LoginController.java:22]:
```suggestion
    public LoginResponse login(@Valid @RequestBody LoginRequest loginRequest) {
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/LoginController.java [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+        return loginService.authenticationOfUser(loginRequest);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/LoginController.java:24]:
```suggestion
        return loginService.authenticateUser(loginRequest);
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+import com.example.event_management.service.UsersService;
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping(path = "/api/users")
+@CrossOrigin(origins = "https://localhost:3000")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:15]: not required

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    @GetMapping(path = "")
+    public List<Users> retrieveAllUsersList() {
+        return usersService.findAll();
+    }
+
+//    //for admin and user
+//    @GetMapping(path = "/username/{username}")
+//    public Users retrieveUserByUsername(@PathVariable String username) {
+//        return usersService.findByUsername(username);
+//    }
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:31]: remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+@RestController
+@RequestMapping(path = "/api/users")
+@CrossOrigin(origins = "https://localhost:3000")
+public class UsersController {
+
+    @Autowired
+    UsersService usersService;
+
+    //for admin
+    @GetMapping(path = "")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:22]:
```suggestion
    @GetMapping
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+
+//    //for admin and user
+//    @GetMapping(path = "/username/{username}")
+//    public Users retrieveUserByUsername(@PathVariable String username) {
+//        return usersService.findByUsername(username);
+//    }
+
+    //for admin
+    @GetMapping(path = "/{id}")
+    public Users retrieveUserById(@PathVariable String id) {
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:35]:
```suggestion
    public Users findById(@PathVariable String id) {
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+        return usersService.findById(id);
+    }
+
+    //for user
+    @PostMapping(path = "/sign-up")
+    public Users signUp(@Valid @RequestBody Users users) {
+        return usersService.save(users);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:42]:
```suggestion
    @PostMapping(path = "/sign-up")
    public Users signUp(@Valid @RequestBody Users users) {
        return usersService.save(users);
```
```suggestion
    @PostMapping
    public Users save(@Valid @RequestBody Users users) {
        return usersService.save(users);
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    //for user and admin
+    @DeleteMapping(path = "/{id}")
+    public void delete(@PathVariable String id) {
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:47]:
```suggestion
    public void deleteById(@PathVariable String id) {
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+        usersService.deleteById(id);
+    }
+
+    //for user
+    @PatchMapping(path = "/{id}")
+    public Users update(@PathVariable(value = "id") String userId, @Valid @RequestBody Users users){
+
+        return usersService.update(userId, users);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:55]:
```suggestion
    public Users updateById(@PathVariable(value = "id") String id, @Valid @RequestBody Users users){
        users.setId(id)
        return usersService.update(users);
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    //for user
+    @PatchMapping(path = "/reset/{id}")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:59]:
```suggestion
    @PatchMapping(path = "/reset-password")
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+    public Users resetPassword(@PathVariable(value = "id") String userId, @Valid @RequestBody ResetPasswordRequest resetPasswordRequest){
+
+        return usersService.resetPassword(userId, resetPasswordRequest);
+    }
+
+
+    //for user
+    @PatchMapping(path = "/forget")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:67]:
```suggestion
    @PatchMapping(path = "/forget-password")
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/ExistingUsernameException.java [hunk 26]
```diff
@@ -0,0 +1,12 @@
+package com.example.event_management.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.CONFLICT)
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/ExistingUsernameException.java:6]: use static import
```suggestion
@ResponseStatus(code = CONFLICT)

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/ExistingUsernameException.java [hunk 27]
```diff
@@ -0,0 +1,12 @@
+package com.example.event_management.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.CONFLICT)
+public class ExistingUsernameException extends RuntimeException{
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/ExistingUsernameException.java:7]:
```suggestion
public class DuplicateUsernameException extends RuntimeException{

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/NotFoundException.java [hunk 28]
```diff
@@ -0,0 +1,13 @@
+package com.example.event_management.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.NOT_FOUND)
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/NotFoundException.java:6]: use static import
```suggestion
@ResponseStatus(code = NOT_FOUND)

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/WrongPasswordException.java [hunk 29]
```diff
@@ -0,0 +1,12 @@
+package com.example.event_management.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.UNAUTHORIZED)
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/exception/WrongPasswordException.java:6]:
```suggestion
@ResponseStatus(code = UNAUTHORIZED)

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java [hunk 30]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    //for user
+    @PostMapping(path = "/status")
+    public BookEvents bookAnEvent(@Valid @RequestBody BookEventRequest bookEventRequest){
+        return eventsService.bookAnEvent(bookEventRequest);
+    }
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:61]:
```suggestion
    @PatchMapping(path = "/{id}/status")
    public BookEvents updateStatusById(@PathVariable String id, @Valid @RequestBody BookEventRequest bookEventRequest){
        String userId = // Fetch user id from user context. Rahul is working on this.
        bookEventRequest.setUserId(userId);
        bookEventRequest.setEventId(id);
        return eventsService.updateStatusById(bookEventRequest);
    }

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/LoginRequest.java [hunk 31]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.*;
+
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class LoginRequest {
+
+    @NotNull(message = "Username can not be empty")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/LoginRequest.java:13]: NotBlank checks for null and blank string as well
```suggestion
    @NotBlank(message = "Username can not be null or blank")
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/LoginRequest.java [hunk 32]
```diff
@@ -0,0 +1,19 @@
...
+    private String username;
+
+    @NotNull(message = "Password can not be empty")
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/LoginRequest.java:16]:
```suggestion
    @NotBlank(message = "Password can not be null or blank")
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/ResetPasswordRequest.java [hunk 33]
```diff
@@ -0,0 +1,14 @@
...
+
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class ResetPasswordRequest {
+
+    private String oldPassword;
+    private String newPassword;
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/ResetPasswordRequest.java:13]: Add NotBlank annotation for both fields with proper error message

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Session.java [hunk 34]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Id
+    @Column(name = "id")
+    private String id;
+    @Column(name = "user_Id")
+    private String userId;
+    @Column(name = "token")
+    private String token;
+    @Column(name = "token_expiry")
+    private String tokenExpiry;
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Session.java:27]:
```suggestion
    @Column(name = "id")
    private String id;
    
    @Column(name = "user_Id")
    private String userId;
    
    @Column(name = "token")
    private String token;
    
    @Column(name = "token_expiry")
    private String tokenExpiry;
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Users.java [hunk 35]
```diff
@@ -0,0 +1,51 @@
...
+    @Column(name = "password")
+    private String password;
+    @Column(name = "full_name")
+    private String fullName;
+    @Column(name = "email")
+    private String email;
+    @Column(name = "mobile_number")
+    private String mobileNumber;
+    @Column(name = "role")
+    private String role;
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Users.java:32]:
```suggestion
    @Column(name = "id")
    private String id;
    
    @Column(name = "username")
    private String username;
    
    @Column(name = "password")
    private String password;
    
    @Column(name = "full_name")
    private String fullName;
    
    @Column(name = "email")
    private String email;
    
    @Column(name = "mobile_number")
    private String mobileNumber;
    
    @Column(name = "role")
    private String role;
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Users.java [hunk 36]
```diff
@@ -0,0 +1,51 @@
...
+
+
+//    @OneToMany(mappedBy = "users")
+//    Set<BookEvents> bookEventsSet;
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Users.java:36]: remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Users.java [hunk 37]
```diff
@@ -0,0 +1,51 @@
...
+
+
+    public void setPassword(String password) {
+
+        BCryptPasswordEncoder bCrypt = new BCryptPasswordEncoder();
+        String encryptedPassword = bCrypt.encode(password);
+        this.password = encryptedPassword;
+
+//        this.password = password;
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/Users.java:45]: remove lines

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/repository/BookEventsRepository.java [hunk 38]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+public interface BookEventsRepository extends JpaRepository<BookEvents, String> {
+    BookEvents findByEventIdAndUserId(String eventId, String userId);
+
+    void deleteByEventId(String id);
+
+//    void deleteByEventId(String eventId);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/repository/BookEventsRepository.java:14]:
```suggestion
    
    BookEvents findByEventIdAndUserId(String eventId, String userId);

    void deleteByEventId(String eventId);

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/repository/EventsRepository.java [hunk 39]
```diff
@@ -0,0 +1,9 @@
+package com.example.event_management.repository;
+
+import com.example.event_management.model.Events;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface EventsRepository extends JpaRepository<Events, String> {
+
+//    Events findByName(String name);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/repository/EventsRepository.java:8]: remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/EventsService.java [hunk 40]
```diff
@@ -0,0 +1,112 @@
...
+        event.setLocation(events.getLocation());
+        event.setTotalSeats(events.getTotalSeats());
+        event.setDescription(events.getDescription());
+
+        return eventsRepository.save(event);
+    }
+
+    public List<BookEvents> findAllBook() {
+        return bookEventsRepository.findAll();
+    }
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/EventsService.java:70]: you should have a different service (UserEventService) to talk with UserEventRepository

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/repository/BookEventsRepository.java [hunk 41]
```diff
@@ -0,0 +1,16 @@
+package com.example.event_management.repository;
+
+import com.example.event_management.model.BookEvents;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface BookEventsRepository extends JpaRepository<BookEvents, String> {
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/repository/BookEventsRepository.java:9]:
```suggestion
public interface UserEventsRepository extends JpaRepository<BookEvents, String> {

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/BookEvents.java [hunk 42]
```diff
@@ -0,0 +1,37 @@
...
+
+@Entity
+@Table(name = "event_status")
+@Builder
+@Getter
+@Setter
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+public class BookEvents {
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/BookEvents.java:14]:
```suggestion
public class UserEvents {

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/model/BookEvents.java [hunk 43]
```diff
@@ -0,0 +1,37 @@
...
+//    public Users users;
+
+    @Column(name = "event_id")
+    private String eventId;
+
+    @Column(name = "user_id")
+    private String userId;
+
+    @Column(name = "status")
+    private String status;
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/model/BookEvents.java:35]:
```suggestion
    @Column(name = "isBookmark")
    private Boolean isBookmark;
    
    @Column(name = "isBooked")
    private Boolean isBooked;
    
    @Column(name = "isAttended")
    private Boolean isAttended;

```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/EventsService.java [hunk 44]
```diff
@@ -0,0 +1,112 @@
...
+                    .eventId(bookEventRequest.getEventId())
+                    .userId(bookEventRequest.getUserId())
+                    .status(bookEventRequest.getEventStatus().getValue())
+                    .build();
+        }
+
+        bookEvent = bookEventsRepository.save(bookEvent);
+
+        return bookEvent;
+    }
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/EventsService.java:110]: - This logic should be part of UserEventService.
- Current method logic is invalid.
- You should three different method to bookmark, booked or mark as attend.

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java [hunk 45]
```diff
@@ -0,0 +1,94 @@
...
+                        .token(session.getToken())
+                        .tokenExpiry(session.getTokenExpiry())
+                        .build();
+
+            }else{
+                throw new WrongPasswordException("Wrong Password");
+           }
+        }
+        else {
+            throw new NotFoundException("Username not found");
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java:59]:
```suggestion
            throw new UnAuthorisedException("Username not found");
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java [hunk 46]
```diff
@@ -0,0 +1,94 @@
...
+                session = sessionRepository.save(session);
+
+                return LoginResponse.builder()
+                        .userId(session.getUserId())
+                        .token(session.getToken())
+                        .tokenExpiry(session.getTokenExpiry())
+                        .build();
+
+            }else{
+                throw new WrongPasswordException("Wrong Password");
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java:55]:
```suggestion
                throw new UnAuthorisedException("Wrong Password");
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java [hunk 47]
```diff
@@ -0,0 +1,94 @@
...
+            if(isPasswordCorrect){
+
+                Session session = Session.builder()
+                        .id(UUID.randomUUID().toString())
+                        .userId(users.getId())
+                        .token(UUID.randomUUID().toString())
+                        .tokenExpiry(Timestamp.from(Instant.now()).toString())
+                        .build();
+
+                session = sessionRepository.save(session);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java:46]: - This should be part of UserSessionService
- method name --> generateUserSessionFor(userId);

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java [hunk 48]
```diff
@@ -0,0 +1,94 @@
...
+//
+//        try {
+//            Date d1 = sdf.parse(now);
+//            Date d2 = sdf.parse(before);
+//
+//            long differenceInTime = d1.getTime() - d2.getTime();
+//
+////            System.out.println(d1 + "     " + d2 + "        " + differenceInTime);
+////            long differenceInSeconds = TimeUnit.MILLISECONDS.toSeconds(differenceInTime) % 60;
+////            System.out.println(differenceInTime);
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java:85]: remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java [hunk 49]
```diff
@@ -0,0 +1,115 @@
...
+    UsersRepository usersRepository;
+
+    static List<Users> usersList = new ArrayList<>();
+    BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+    boolean isPasswordCorrect = false;
+
+//    private static int usersCount = 1000;
+//        static {
+//        usersList.add(new Users(++usersCount, "Gajendra", "G", "J", "email", "123", 432, "user"));
+//    }
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java:31]: - remove global variables if not required
- remove commented line

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java [hunk 50]
```diff
@@ -0,0 +1,115 @@
...
+    public List<Users> findAll() {
+        return usersRepository.findAll();
+    }
+
+    public Users findByUsername(String username) {
+        Users user = usersRepository.findByUsername(username);
+
+        if(Objects.nonNull(user)){
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java:39]:
```suggestion
        if(Objects.isNull(user)){
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java [hunk 51]
```diff
@@ -0,0 +1,115 @@
...
+            throw new NotFoundException("Username not found");
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java:40]:
```suggestion
            throw new NotFoundException("User not found");
```

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java [hunk 52]
```diff
@@ -0,0 +1,115 @@
...
+
+    public void deleteById(String id){
+        usersRepository.deleteById(id);
+    }
+
+    public Users update(String id, Users users) {
+        Users updateUser = usersRepository.findById(id).orElseThrow(() -> new NotFoundException("User Id not found"));
+
+        updateUser.setFullName(users.getFullName());
+        updateUser.setMobileNumber(users.getMobileNumber());
```
Comment (sagarzond) [inline Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java:77]: why only set these two fields?

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java (hunk 30 above)
Comment (sagarzond) [reply Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:61]: - This method should be part of UserEventController.
- Update bookmark --> PATCH /user/{id}/events/{id}/bookmark?isBookmark=true/false
- Update booked status --> PATCH /user/{id}/events/{id}/booked?isBooked=true/false
- Update attended status --> PATCH /user/{id}/events/{id}/attended?isAttended=true/false

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java (hunk 52 above)
Comment (sagarzond) [reply Event-Management-Application/event_management/src/main/java/com/example/event_management/service/UsersService.java:77]: This is not required

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java (hunk 11 above)
Comment (gajendraJangid421) [reply Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/EventsController.java:48]: Here, its first finding whether event id is present or not and then its updating.

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java (hunk 23 above)
Comment (gajendraJangid421) [reply Event-Management-Application/event_management/src/main/java/com/example/event_management/controller/UsersController.java:55]: Here, its first finding whether user id is present or not and then its updating.

File: Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java (hunk 48 above)
Comment (gajendraJangid421) [reply Event-Management-Application/event_management/src/main/java/com/example/event_management/service/LoginService.java:85]: needed for token for session expiry

**Pull request merged**